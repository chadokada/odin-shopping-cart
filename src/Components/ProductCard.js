import React from 'react';


const ProductCard = ({product}) => {
  
  return(
    <div className='product-container'>
      <div className='product-img-container'>
        <img 
          className='product-img'
          alt={product.name} 
          src={require(`../Images/Products/${product.source}`)}
        />
      </div>

      <div className='product-info'>
        <div className='product-name'>
          {product.name}
        </div>
        <div className='price'>
          {`$${product.price.toFixed(2)}`}
        </div>
      </div>

      <button className='add-to-cart-btn'>Add To Cart</button>
    </div>
  )
}

export default ProductCard;