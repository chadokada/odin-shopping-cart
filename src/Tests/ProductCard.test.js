import React from "react";
import { render, screen } from '@testing-library/react';
import userEvent from "@testing-library/user-event";
import ProductCard from '../Components/ProductCard';

describe("Test ProductCard", () => {
  const testProduct = {
    name: 'Bowler Hat',
    type: 'Hat',
    price: 15,
    source: 'bowler.jpg'
  };
  const onClickMock = jest.fn();

  it("renders price with two decimals", () => {
    render(<ProductCard product={testProduct} handleAddToCart={onClickMock} />);

    expect(
      screen.getByText(
        (content, element) => content.startsWith('$')).textContent)
        .toMatch('$15.00');
  });

  it("add to cart callback called on correct number of times", () => {
    render(<ProductCard product={testProduct} handleAddToCart={onClickMock} />);
    const addToCart = screen.getByRole('button', {name: 'Add To Cart'});

    userEvent.click(addToCart);
    
    expect(onClickMock).toHaveBeenCalledTimes(1);
  })


  it("productcard snapshot", () => {
    const {container} = render(<ProductCard product={testProduct} handleAddToCart={onClickMock} />);

    expect(container).toMatchSnapshot();
  });


})