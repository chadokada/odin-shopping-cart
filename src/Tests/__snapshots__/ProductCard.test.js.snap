// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProductCard productcard snapshot 1`] = `
<div>
  <div
    class="product-container"
    name="Bowler Hat"
    type="Hat"
  >
    <div
      class="product-img-container"
    >
      <img
        alt="Bowler Hat"
        class="product-img"
        src="bowler.jpg"
      />
    </div>
    <div
      class="product-info"
    >
      <div
        class="product-name"
      >
        Bowler Hat
      </div>
      <div
        class="price"
      >
        $15.00
      </div>
    </div>
    <button
      class="add-to-cart-btn"
    >
      Add To Cart
    </button>
  </div>
</div>
`;
